client/implementation/src/main/java/io/smallrye/graphql/client/impl/GraphQLClientsConfiguration.java

```diff
@@ -55,40 +60,32 @@ public GraphQLClientsConfiguration() {
         }
         for (String clientName : detectedClientNames) {
             GraphQLClientConfiguration configuration = new GraphQLClientConfiguration();
-            configuration.setUrl(mpConfig.getValue(clientName + "/mp-graphql/url", String.class));
-            configuration.setHeaders(getConfigurationValueMap(clientName, "header", mpConfig));
+            mpConfig.getOptionalValue(clientName + "/mp-graphql/url", String.class)
+                    .ifPresent(configuration::setUrl);
+            configuration.setHeaders(getConfiguredHeaders(clientName, mpConfig));
             clients.put(clientName, configuration);
         }
     }
 
     /**
-     * Scan the passed Java interfaces for `@GraphQLClientApi` annotations and create and register
-     * client configuration objects for them. This needs to be called by the runtime some time during initialization,
+     * Scan the passed Java interface for `@GraphQLClientApi` annotations and create and register
+     * client configuration objects for it. This needs to be called by the runtime some time during initialization,
      * before clients are actually created.
      */
-    public void addTypesafeClientApis(List<Class<?>> apis) {
-        if (apis == null) {
-            SmallRyeGraphQLClientLogging.log.apisNotSet();
-            apis = Collections.emptyList();
-        }
-        for (Class<?> api : apis) {
-            TypesafeClientConfigurationReader configReader = new TypesafeClientConfigurationReader(api);
-            clients.put(configReader.getConfigKey(), configReader.getClientConfiguration());
-        }
+    public void initTypesafeClientApi(Class<?> api) {
+        TypesafeClientConfigurationReader reader = new TypesafeClientConfigurationReader(api);
+        clients.put(reader.getConfigKey(), reader.getClientConfiguration());
     }
 
     public GraphQLClientConfiguration getClient(String key) {
         return clients.get(key);
     }
 
-    public void addClient(String key, GraphQLClientConfiguration config) {
```

Comment:
We have to keep this, this is called by Quarkus
```
[ERROR] /home/jmartisk/Workspace/quarkus/extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientConfigurationMergerBean.java:[50,38] cannot find symbol
[ERROR]   symbol:   method addClient(java.lang.String,io.smallrye.graphql.client.impl.GraphQLClientConfiguration)
[ERROR]   location: variable upstreamConfiguration of type io.smallrye.graphql.client.impl.GraphQLClientsConfiguration
```

Reply:
done